src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,46 @@
-import React from 'react';
+import React, { useContext, useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { TodosContext } from '../TodosContext/TodosContext';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  currentUserId: number | null,
+  todos: Todo[] | null,
+};
+
+export const TodoModal: React.FC<Props> = ({
+  currentUserId, todos,
+}) => {
+  const { handleShowModal, currentTodoId } = useContext(TodosContext);
+
+  const [currentUser, setCurrentUser] = useState<User | null>(null);
+  const [currentTodo, setCurrentTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    if (currentUserId) {
+      getUser(currentUserId).then((user) => {
+        setCurrentUser(user);
+      });
+
+      let currTodo;
+
+      if (todos !== null) {
+        currTodo = todos.find(todo => todo.id === currentTodoId);
+      }
```

Comment:
```suggestion

     
        const currTodo = todos.length ? todos.find(todo => todo.id === currentTodoId) : null ;
    
```

If you can write constant instead let use it prefer. 
let worsens the readability and perception of the code during scaling